pkg/server/api/entry/v1/service.go

```diff
@@ -658,6 +755,13 @@ func (s *Service) updateEntry(ctx context.Context, e *types.Entry, inputMask *ty
 			Hint:          inputMask.Hint,
 		}
 	}
+
+	if err := validateRegistrationEntryForUpdate(convEntry, mask); err != nil {
```

Comment:
An alternative to this would be to add all the validations here in https://github.com/spiffe/spire/blob/3d158ce5635ab92fc5869b4ecda44c6a153b2a06/pkg/server/api/entry.go#L92 but I wasn't sure that's where the maintainers would want to add these checks.

That function already does some validations, but mostly as a necessity of handling errors. It would also simplify things a bit. Let me know if that's better and I can change this.